tower/src/make/make_service.rs

```diff
@@ -65,3 +147,51 @@ where
         Service::call(self, target)
     }
 }
+
+#[derive(Debug, Clone)]
+pub struct IntoService<M, Request> {
+    make: M,
+    _marker: PhantomData<Request>,
+}
+
+impl<M, S, Target, Request> Service<Target> for IntoService<M, Request>
+where
+    M: Service<Target, Response = S>,
+    S: Service<Request>,
+{
+    type Response = M::Response;
+    type Error = M::Error;
+    type Future = M::Future;
+
+    fn poll_ready(&mut self, cx: &mut Context<'_>) -> Poll<Result<(), Self::Error>> {
+        self.make.poll_ready(cx)
+    }
+
+    fn call(&mut self, target: Target) -> Self::Future {
+        self.make.make_service(target)
+    }
+}
+
+#[derive(Debug)]
+pub struct AsService<'a, M, Request> {
```

Comment:
similarly, a quick doc comment would be good
```suggestion
/// Service returned by [`MakeService::as_service`][as]. 
/// 
/// See the documentation on [`as_service`][as] for details.
///
/// [as]: MakeService::as_service
#[derive(Debug)]
pub struct AsService<'a, M, Request> {
```